src/semver-intersect.js

```diff
@@ -95,42 +108,63 @@ function formatIntersection ({ lowerBound = '', upperBound = '' }) {
     return `${lowerBound} ${upperBound}`.trim();
 }
 
-function intersect (...ranges) {
-    ranges = expandRanges(...ranges);
+function updateBounds({ lowerBound, upperBound }, range) {
```

Comment:
This function is totally the same as the callback argument of reduce function in the parent revision